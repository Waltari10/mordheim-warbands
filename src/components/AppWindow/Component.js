import React from 'react';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles(theme => ({
  container: {


    [theme.breakpoints.down('sm')]: {
      width: '100%',
      maxWidth: '100% !important',
      height: '100%',
      borderRadius: 0,
      minHeight: '100%',
      padding: 0,
    },

    // padding: theme.spacing(6),
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '35%',
    minHeight: '70%',
    overflow: 'hidden',
    maxHeight: '70%',
    overflowY: 'auto',
  },
}));

const AppWindow = ({ className, children, size }) => {

  const classes = useStyles();

  let width;

  if (size === 'xs') {
    width = '400px';
  } else {
    width = '600px';
  }

  return (
    <Paper
      style={{
        maxWidth: width,
      }}
      className={`${classes.container} ${className}`}
    >
      {children}
    </Paper>
  );
};
 
export default AppWindow;