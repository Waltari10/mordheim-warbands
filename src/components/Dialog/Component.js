import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles(() => ({
  paper: {
    backgroundColor: 'white',
  },
}));

export default function MyDialog({
  handleClose, open, title, dialog, cancel, confirm,
  handleConfirm,
}) {

  const classes = useStyles();

  return (
    <Dialog
      open={open}
      onClose={handleClose}
      classes={{
        paper: classes.paper,
      }}
    >
      <DialogTitle>{title}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description">
          {dialog}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button
          onClick={handleClose}
          color="primary"
        >
          {cancel || 'Cancel'}
        </Button>
        <Button
          onClick={handleConfirm}
          color="primary"
          autoFocus

        >
          {confirm || 'Confirm'}
        </Button>
      </DialogActions>
    </Dialog>
  );
}