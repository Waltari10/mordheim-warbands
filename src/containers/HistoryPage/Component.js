import React from 'react';

// TODO: Add indicator indicating how far user is in reflcetion
// Add 5xwhy


const HistoryPage = () => {
  return (<div>HistoryPage</div>);
};
 
export default HistoryPage;