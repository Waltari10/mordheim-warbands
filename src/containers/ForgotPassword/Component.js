import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Link from '@material-ui/core/Link';
import InputAdornment from '@material-ui/core/InputAdornment';
import { makeStyles } from '@material-ui/core/styles';
import { Link as RouterLink, Redirect } from 'react-router-dom';
import EmailIcon from '@material-ui/icons/EmailOutlined';
import InputBase from '@material-ui/core/InputBase';
import * as constants from '../../constants';

import AppWindow from '../../components/AppWindow';

const useStyles = makeStyles((theme) => ({
  header: {
    marginTop: theme.spacing(0),
    marginBottom: theme.spacing(12),
  },
  email: {
    background: 'white',
    borderTopRightRadius: '4px',
    borderTopLeftRadius: '4px',
    height: '48px',
    overflow: 'hidden',
    borderWidth: '0.5px',
    borderStyle: 'solid',
    borderColor: '#C0C0C0',
  },
  continueButton: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(3),
  },
  loginLink: {
    marginBottom: theme.spacing(3),
  },
  content: {
    width: '100%',
  },
  errorMsg: {
    marginBottom: theme.spacing(3),
  },
  inputRoot: {
    backgroundColor: 'white',
    height: '48px',
  },
  input: {
    height: '48px',
    padding: '27px 12px 24px',
  },
  inputAdornment: {
    color: theme.palette.primary.main,
  },
  inputStartAdornment: {
    color: theme.palette.primary.main,
    marginLeft: theme.spacing(2),
  },
}));


const Login = ({
  isAuthorized, sendResetPasswordEmailRequestState, sendResetPasswordEmail,
}) => {

  const [email, setEmail] = useState('');

  const isDisabled = (
    sendResetPasswordEmailRequestState === constants.LOADING ||
    sendResetPasswordEmailRequestState === constants.SUCCESS
  );

  const submitDisabled = isDisabled || !email;

  const classes = useStyles();

  return (
    <AppWindow size="xs">

      {
        isAuthorized &&
          <Redirect
            to='/'
          />
      }

      <Grid className={classes.content} alignItems="center" direction="column" container>
        <Grid className={classes.header} item>
          <Typography variant="h4">Mordheim Warbands</Typography>
        </Grid>
        <Grid item>
          <InputBase
            style={{ display: 'flex' }}
            className={classes.email}
            onChange={(e) => setEmail(e.target.value)}
            // label="Type your email"
            disabled={isDisabled}
            placeholder="email"
            value={email}
            variant="filled"
            startAdornment={
              <InputAdornment position="start">
                <EmailIcon className={classes.inputStartAdornment} />
              </InputAdornment>
            }
          />
        </Grid>
        <Grid item>
          <Button
            className={classes.continueButton}
            disabled={submitDisabled}
            onClick={() => sendResetPasswordEmail(email)}
            variant="contained"
            color="primary"
          >Send password reset email</Button>
        </Grid>

        {
          sendResetPasswordEmailRequestState === constants.ERROR && (
            <Grid className={classes.errorMsg} item>
              <Typography color="error" variant="body1">Something went wrong!</Typography>
            </Grid>
          )
        }
        <Grid
          className={classes.loginLink}
          item
        >
          <Link
            component={RouterLink}
            to={'/'}
          >
            <Typography variant="body2">
              Go back to login
            </Typography>
          </Link>
        </Grid>
      </Grid>
    </AppWindow>
  );
};
 
export default Login;