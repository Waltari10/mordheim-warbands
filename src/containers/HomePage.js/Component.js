import React, { useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Progress from '@material-ui/core/CircularProgress';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Menu from '@material-ui/core/Menu';
import { Link as RouterLink } from 'react-router-dom';
import MenuItem from '@material-ui/core/MenuItem';
import MenuIcon from '@material-ui/icons/Menu';
import IconButton from '@material-ui/core/IconButton';
import { Typography } from '@material-ui/core';

import PrivateRoute from '../PrivateRoute';
import ReflectionModal from '../ReflectionModal';

import ReflectionListItem from './ReflectionListItem';
import AppWindow from '../../components/AppWindow';


const useStyles = makeStyles((theme) => ({
  listItem: {
    
  },
  window: {
    padding: 0,
    position: 'relative',
    flexDirection: 'column',
    overflow: 'hidden',
  },
  container: {
    marginTop: '53px',
    width: '100%',
    height: '100%',
    placeSelf: 'flex-start',
    overflowY: 'scroll',
    overflowX: 'hidden',
    flex: 1,
  },
  progress: {
    position: 'absolute',
    top: 'calc(50% - 10px)',
    left: 'calc(50% - 10px)',
  },
  hero: {
    position: 'absolute',
    right: theme.spacing(3),
    bottom: theme.spacing(3),
  },
  addIcon: {
    color: 'white',
    marginTop: '6px',
  },
  topNavigationContainer: {
    boxShadow: '0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14), 0px 1px 8px 0px rgba(0,0,0,0.12)',
    height: '53px',
    position: 'absolute',
    width: '100%',
    zIndex: 1,
    left: 0,
    right: 0,
    backgroundColor: '#9e9e9e',
    top: 0,
  },
  title: {
    lineHeight: '53px',
  },
  menuIcon: {

    position: 'absolute',
    top: '5px',
    right: theme.spacing(3),
  },
}));


const HomePage = ({
  isLoading, reflections = {}, getReflections, match, logout, uid,
  
}) => {

  useEffect(() => {
    if (uid) {
      getReflections();
    }
  }, [uid]);


  const classes = useStyles();

  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <AppWindow className={classes.window} size="xs">

      <div
        className={classes.topNavigationContainer}
      >
        <Typography className={classes.title} align="center" variant="h6">
            Reflection log
        </Typography>


        <IconButton
          className={classes.menuIcon}
          onClick={handleClick}
        >
          <MenuIcon />
        </IconButton>

        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          keepMounted
          open={Boolean(anchorEl)}
          onClose={handleClose}
        >
          <MenuItem onClick={() => {
            handleClose();
            logout();
          }}>Logout</MenuItem>
        </Menu>

        
      </div>


      <div className={classes.container}>
        <PrivateRoute component={ReflectionModal} path="/reflection/:id" />
        {isLoading && <Progress className={classes.progress} />}
        {!isLoading && Object.keys(reflections).map((key) => {
          const reflection = reflections[key];

          return (
            <ReflectionListItem
              match={match}
              key={key}
              id={key}
              reflection={reflection}
            />
          );
        })}

        {!isLoading && Object.keys(reflections).length === 0 && (
          <Typography>No reflections yet. Add first reflection.</Typography>
        )}

        
      </div>


      <Fab
        className={classes.hero}
        color="primary"
      >
        <RouterLink
          to={`${match.path}reflection/new`}
        >
          <AddIcon className={classes.addIcon}/>
        </RouterLink>
      </Fab>
    </AppWindow>
  );
};
 
export default HomePage;