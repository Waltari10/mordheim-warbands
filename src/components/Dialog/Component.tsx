import React, { memo } from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles(() => ({
  paper: {
    backgroundColor: 'white',
  },
}));

interface Props {
  handleClose(): void;
  open?: boolean;
  title?: string;
  dialog?: any;
  cancel?: String;
  confirm?: string;
  handleConfirm(): void;
  isConfirm?: boolean;
  isCancel?: boolean;
}

const MyDialog = memo(({
  handleClose, open = false, title, dialog, cancel, confirm,
  handleConfirm, isConfirm = true, isCancel = true,
}: Props) => {

  const classes = useStyles();

  return (
    <Dialog
      open={open}
      onClose={handleClose}
      classes={{
        paper: classes.paper,
      }
      }
    >
      <DialogTitle>{title} </DialogTitle>
      <DialogContent>
        <DialogContentText
          id="alert-dialog-description"
          dangerouslySetInnerHTML={{ __html: dialog }}
        >
          {/* {dialog} */}
        </DialogContentText>
      </DialogContent>
      < DialogActions>
        {isCancel && <Button
          onClick={handleClose}
          color="primary"
        >
          {cancel || 'Cancel'}
        </Button>}
        {
          isConfirm && <Button
            onClick={handleConfirm}
            color="primary"
            autoFocus

          >
            {confirm || 'Confirm'}
          </Button>}
      </DialogActions>
    </Dialog>
  );
});

export default MyDialog;