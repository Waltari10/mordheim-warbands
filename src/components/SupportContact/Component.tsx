import React from 'react';
import Dialog from '../Dialog';

interface Props {
  handleClose(): void;
  handleConfirm?(): void;
  open: boolean;
}

const SupportContact: React.FunctionComponent<Props> = (props) => {

  return (
    <Dialog
      title={'Support/Contact'}
      confirm="Close"
      dialog={`
      
      <h4>Support</h4>
      If you are experiencing problems, or have any questions, suggestions or 
      comments please contact me at 
      <a href="mailto:warband.manager@gmail.com">warband.manager@gmail.com</a>.<br><br>


      You can also say hi to me on Twitter. I'd love to hear your opinion on Warband Manager! 
      <a href="https://twitter.com/ValdeCode" target="_blank">@valdeCode<a/><br><br>

      <h4>Want to help?</h4>
      I'd be happy to share the source code, if somebody want's to help by making a pull request. There are a ton of 
      features that could be added and so little time. So any help would be appreciated.<br>
      Right now the tech stack includes: React, Redux, AWS and a few others.
      `}
      isConfirm={false}
      {...props}
    />
  );
};

export default SupportContact;


// <br><br>
//       <h4>Donations</h4>
//       This project is not going to make any money. I just wanted to make it because
//       I had a hard time maintaing my roster. Still having a domain, database and hosting costs money
//       after a certain point, so I might have to start asking for donations or figure out other ways to maintain
//       this. For now, well see how far the free plan on AWS take us.