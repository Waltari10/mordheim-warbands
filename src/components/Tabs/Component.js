import React from 'react';
// import { makeStyles } from '@material-ui/core/styles';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import PlusIcon from '@material-ui/icons/Add';
import HistoryIcon from '@material-ui/icons/History';
import AppBar from '@material-ui/core/AppBar';
import PersonPinIcon from '@material-ui/icons/PersonPin';
import { Route, Switch, Link } from 'react-router-dom';
import ReflectPage from '../../containers/ReflectPage';
import HistoryPage from '../../containers/HistoryPage';
import UserPage from '../../containers/UserPage';

// const useStyles = makeStyles({
//   root: {
//     flexGrow: 1,
//     maxWidth: 500,
//   },
// });


const MyTabs = ({ match }) => {
  // const classes = useStyles();
  const [value, setValue] = React.useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  console.log('tabs');
  return (
    <>

      <div
        style={{
          height: 'calc(100% - 72px)',
          width: '100%',
          position: 'relative',
        }}
      >
        <Switch>
          <Route path={'/history'} component={HistoryPage}/>
          <Route path={'/reflect'} component={ReflectPage}/>
          <Route path={'/user'} component={UserPage}/>
        </Switch>
      </div>
      <AppBar
        color="transparent"
        position="static"
        style={{
          position: 'absolute',
          bottom: 0,
          left: 0,
          right: 0,
        }}
      >
        <Tabs
          value={value}
          onChange={handleChange}
          variant="fullWidth"
          indicatorColor="secondary"
          textColor="primary"
        >
          <Tab
            to={`${match.url}history`}
            component={Link}
            icon={<HistoryIcon />}
            label="HISTORY"
          />
          <Tab
            to={`${match.url}reflect`}
            component={Link}
            icon={<PlusIcon />}
            label="REFLECT" />
          <Tab
            to={`${match.url}user`}
            component={Link}
            icon={<PersonPinIcon />}
            label="USER" />
        </Tabs>
      </AppBar>
    </>
  );
};
 
export default MyTabs;