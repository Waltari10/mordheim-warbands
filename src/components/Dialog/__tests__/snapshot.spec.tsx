import Component from '../index';

import React from 'react';
import { render } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
// eslint-disable-next-line no-unused-vars
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';


const theme = createMuiTheme({});


describe('Dialog', () => {
  it('Should render (smoke test)', () => {
    const { asFragment } = render(
      // Requires to be wrapped in router because NotFoundPage uses react-router-doms Link element
      // Requires to be wrapped MuiThemeProvider because NotFoundPage is using Theme from material-ui
      <Router>
        <MuiThemeProvider theme={theme} >
          <Component
            handleClose={() => { }}
            handleConfirm={() => { }}
          />
        </MuiThemeProvider>
      </Router>,
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});