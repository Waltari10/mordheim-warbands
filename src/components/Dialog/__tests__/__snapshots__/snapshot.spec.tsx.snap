// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Should render (smoke test) 1`] = `<DocumentFragment />`;
